src/components/checkbox/checkbox.test.js

```diff
@@ -0,0 +1,557 @@
+  });
+
+  it.each([
+    [true, "be.disabled"],
+    [false, "not.be.disabled"],
+  ])(
+    "should render Checkbox component with disabled prop set to %s",
+    (booleanValue, assertion) => {
+      CypressMountWithProviders(<CheckboxComponent disabled={booleanValue} />);
+      checkboxRole().should(assertion);
+    }
+  );
+
+  it("should render Checkbox component with id", () => {
+    CypressMountWithProviders(
+      <CheckboxComponent id="f62d5e6e-248c-4b04-ae51-f7fab5bd93a3" />
+    );
+    checkboxRole().should("have.id", "f62d5e6e-248c-4b04-ae51-f7fab5bd93a3");
+  });
+
+  it("should render Checkbox component with name", () => {
+    CypressMountWithProviders(<CheckboxComponent name="confirm permission" />);
+    checkboxRole().should("have.attr", "name", "confirm permission");
+  });
+
+  it("should render Checkbox component with value prop", () => {
+    CypressMountWithProviders(<CheckboxComponent value="checkboxvalue" />);
+    checkboxRole().should("have.value", "checkboxvalue");
+  });
+
+  it.each([
+    ["small", "16"],
+    ["large", "24"],
+  ])("should render Checkbox component with size set to %s", (size, width) => {
+    CypressMountWithProviders(<CheckboxComponent size={size} />);
+    checkboxRole().should("have.css", "height").and("contain", `${width}px`);
+    checkboxRole().should("have.css", "width").and("contain", `${width}px`);
+  });
+
+  it.each([
+    [1, "8px"],
+    [2, "16px"],
+  ])(
+    "should render Checkbox component with %s as labelSpacing",
+    (spacing, padding) => {
+      CypressMountWithProviders(<CheckboxComponent labelSpacing={spacing} />);
+
+      label().parent().should("have.css", "padding-left", padding);
+    }
+  );
+
+  it.each([
+    ["10", "90", "135px", "1215px"],
+    ["30", "70", "405px", "945px"],
+    ["80", "20", "1080px", "270px"],
+  ])(
+    "should render Checkbox using %s as labelWidth, %s as inputWidth and render it with correct label and input width ratios",
+    (labelWidth, inputWidth, labelRatio, inputRatio) => {
+      CypressMountWithProviders(
+        <CheckboxComponent
+          labelInline
+          labelWidth={labelWidth}
+          inputWidth={inputWidth}
+        />
+      );
+
+      label().parent().should("have.css", "width", labelRatio);
+      checkboxRole().parent().should("have.css", "width", inputRatio);
+    }
+  );
+
+  it("should render Checkbox component as a required field", () => {
+    CypressMountWithProviders(
+      <CheckboxComponent label="Required Checkbox" required />
+    );
+    verifyRequiredAsteriskForLabel();
+  });
+
+  it("should render Checkbox component with autoFocus", () => {
+    CypressMountWithProviders(<CheckboxComponent autoFocus />);
+    checkboxRole().should("be.focused");
+  });
+
+  it("should render Checkbox component with error ", () => {
+    CypressMountWithProviders(<CheckboxComponent error />);
+
+    cy.get("svg").should("have.css", "border-bottom-color", "rgb(205, 56, 75)");
```

Comment:
the same here - we need to get `icon()`

Reply:
Updated however it doesn't work on icon(), I have to verify the svg element